resource/ui/ts/pages/stores.ts

```diff
@@ -79,48 +73,71 @@ module AdminViews {
          */
         export module StorePage {
             class Controller {
-                charts: QueryHolder[] = [];
+                exec:metrics.Executor;
+                axes: metrics.Axis[] = [];
+                private _query:metrics.Query;
                 private static _queryEveryMS = 10000;
                 private _interval: number;
                 private _storeId: string;
 
                 private _refresh():void {
                     storeStatuses.refresh();
-                    for (var i = 0; i < this.charts.length; i++) {
-                        this.charts[i].Result.refresh();
-                    }
+                    this.exec.refresh();
                 }
 
-                private _addChart(q:metrics.Query):void {
-                    this.charts.push({
-                        Query: q, 
-                        Result: new Utils.QueryCache(q.execute),
-                    });
+                private _addChart(axis:metrics.Axis):void {
+                    axis.selectors().forEach((s) => this._query.selectors().push(s));
+                    this.axes.push(axis);
                 }
 
                 public constructor(storeId:string) {
                     this._storeId = storeId;
+                    this._query = metrics.NewQuery();
                     this._addChart(
-                        metrics.NewQuery(
-                            metrics.select.Avg(_storeMetric(storeId, "keycount")))
-                        .title("Key Count"));
+                        metrics.NewAxis(
```

Comment:
I expect these pages to change considerably as work continues, I'm not sure where the list of charts to display will end up.  I don't want to spend too much time removing boilerplate.